examples/seq2seq/seq2seq_trainer.py

```diff
@@ -62,9 +81,19 @@ def create_optimizer_and_scheduler(self, num_training_steps: int):
                 )
 
         if self.lr_scheduler is None:
-            self.lr_scheduler = get_linear_schedule_with_warmup(
+            self.lr_scheduler = self._get_lr_scheduler(num_training_steps)
+
+    def _get_lr_scheduler(self, num_training_steps):
+        get_schedule_func = arg_to_scheduler[self.args.lr_scheduler]
+        if self.args.lr_scheduler not in ["constant", "constant_w_warmup"]:
```

Comment:
(nit) slightly cleaner to check positive conditions

if constant: ...
elif constant_w_warmup: ...
else: ...